src/llm/language_model/legacy/legacy_executor.cpp

```diff
@@ -47,7 +49,7 @@ void LegacyExecutor::waitForRequests(std::atomic<bool>* receivedEndSignal) {
 }
 
 void LegacyExecutor::addRequest(std::shared_ptr<LegacyServableExecutionContext> request) {
-    std::lock_guard<std::mutex> guard(queueMutex);
+    std::unique_lock<std::mutex> lock(queueMutex);
```

Comment:
why using unique lock? lock guard is sufficient in this case since we never need to unlock manually